File: practice/algorithms/implementation/modified_kaprekar_numbers/modified_kaprekar_numbers.py
def is_kaprekar(i_n):
    d = len(str(i_n))
    s = i_n ** 2
    r = s % 10 ** d
    l = s // 10 ** d
    return r + l == i_n

p = int(raw_input())
q = int(raw_input())

res = []
for VAL in range(p, q + 1):
    if is_kaprekar(VAL):
        res.append(VAL)
if res:
    for VAL in res:
        print VAL, 
else:
    print 'INVALID RANGE'

File: practice/algorithms/implementation/modified_kaprekar_numbers/modified_kaprekar_numbers.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

bool isKaprekar(int value)
{
    int c = value;
    int cnt = 0;
    while (c > 0)
    {
        cnt++;
        c /= 10;
    }
    
    long long sqr = pow(value, 2);
    long long base = pow(10, cnt);
    
    long long right = sqr % base;
    long long left = sqr / base;
    
    return (left + right) == value;
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */  
    int p, q;
    cin >> p >> q;
    bool hasKaprekar = false;
    for (int i = p; i < q + 1; i++)
    {
        if (isKaprekar(i))
        {
            hasKaprekar = true;
            cout << i << " ";
        }
    }
    if (!hasKaprekar)
    {
        cout << "INVALID RANGE";
    }
    return 0;
}

